scripts/azureml-assets/azureml/assets/model/registry_utils.py

```diff
@@ -149,6 +229,18 @@ def prepare_model(spec_path, model_config, temp_dir, ml_client: MLClient, copy_u
         return None, False
 
 
+def prepare_data(spec_path, data_config, temp_dir, ml_client: MLClient, copy_updater: CopyUpdater = None):
+    """Prepare data for publish."""
+    try:
+        registry_name = ml_client.data._registry_name
+        data_asset = DataAsset(spec_path, data_config, registry_name, temp_dir, copy_updater)
+        data = data_asset.prepare_data(ml_client)
+        return data, True
+    except Exception as e:
+        logger.log_error(f"prepare data failed for {spec_path}. Error {e}")
+        return None, False
+
+
 def update_model_metadata(
```

Comment:
should we keep just this function here and move the RegistryUtils and model/data copying to a new folder?

Reply:
However you want to do it will be fine. Just make sure references are updated. Here's [one example in azureml-asset](https://msdata.visualstudio.com/Vienna/_git/azureml-asset?path=/msftkube/scripts/update_assets.py&version=GBmain&line=11&lineEnd=12&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).

Reply:
keeping as-is for now!